Dockerfile

```diff
@@ -7,5 +7,8 @@ RUN go install --tags extended
 WORKDIR /www
 RUN git clone https://github.com/matcornic/hugo-theme-learn/ themes/learn
 COPY . /www/
-ENTRYPOINT ["hugo", "server", "--bind", "0.0.0.0"]
-CMD [""]
\ No newline at end of file
+RUN /go/bin/hugo --destination /usr/share/nginx/html
+CMD ["hugo", "server", "--bind", "0.0.0.0"]
+
+FROM nginx:alpine
+COPY --from=build /usr/share/nginx/html /usr/share/nginx/html
```

Comment:
I guess we have to different purposes here. 
  1. Use a containerized environment to change the content locally and have the docker container update the website on the fly
  2. create an artifact of the website that is frozen and allow people to consume this artifact without caring about the actual source content.

I changed the `Dockerfile` so that is has even more steps.
  1. *hugo* Creates the plain Hugo image, without adding the content. Thus, it can be used in the `docker-compose` service, as we will map in the content as volume mounts anyway
  2. *build* adds the content and runs a build to populate `/output`
  3. The final image will just copy the static website into a NGINX image to serve it without building it again.

Within the docker-compose file I remove the `image:` statement as it is ignored by the `build` statement anyways.
Furthermore I instruct the compose stack to use `target: hugo` which will use the plain hugo image instead of using the resulting NGINX one.